### PR TITLE
Cscmetax 303 oaipmh sets

### PR DESCRIPTION
**Pull request by jkesanie**
This implementation adds multiple urns to a single record, which means that urn resolver must check for those (should not be a problem).

### CODE REVIEW

File: src/metax_api/api/oaipmh/base/metax_oai_server.py [hunk 1]
```diff
@@ -39,15 +39,28 @@ def _get_filtered_records(self, set, cursor, batch_size, from_=None, until=None)
...
-    def _get_oai_dc_urnresolver_metadata(self, source_identifier, target_identifier):
+    def _get_oai_dc_urnresolver_metadata(self, record):
+        """
+        Preferred identifier is added only for ida and att catalog records
+        other identifiers are added for all.
+        """
+        identifiers = []
+        identifiers.append(settings.OAI['ETSIN_URL_TEMPLATE'] % record.urn_identifier)
+        if record.catalog_versions_datasets():
+            identifiers.append(record.urn_identifier)
```
Comment (hannu40k) [inline src/metax_api/api/oaipmh/base/metax_oai_server.py]: Was the plan to export (unique) preferred_identifiers only, and no urn_identifiers (metadata_version_identifier) ? Or is this code relevant for that part

Comment (jkesanie) [reply src/metax_api/api/oaipmh/base/metax_oai_server.py]: This is implemented according to the current versioning scheme. I will change it when urn_identifier field is removed.

Comment (hannu40k) [reply src/metax_api/api/oaipmh/base/metax_oai_server.py]: I mean in general. preferred_identifier should be the one that is exported?

Comment (jkesanie) [reply src/metax_api/api/oaipmh/base/metax_oai_server.py]: Hmmm...You're right. 
What is being exported is one url and n urns. And urns should come from preferred identifier and other identifiers. I'm going to change that piece of code.

**Comment by jkesanie** [comment]
preferred identifier is handled only for catalog that are not harvested, so they should only include att urns (at least for now).

File: src/metax_api/api/oaipmh/base/metax_oai_server.py [hunk 2]
```diff
@@ -50,10 +50,12 @@ def _get_oai_dc_urnresolver_metadata(self, record):
         Preferred identifier is added only for ida and att catalog records
         other identifiers are added for all.
         """
+        preferred_identifier = record.research_dataset.get('preferred_identifier')
```
Comment (hannu40k) [inline src/metax_api/api/oaipmh/base/metax_oai_server.py]: Does not warrant a change, but for future convenience: preferred_identifier can be accessed like record.preferred_identifier in a similar way like you used urn_identifier before.

**Comment by hannu40k** [comment]
OK

**Pull request merged**